lib/stellar-core-backup/utils.rb

```diff
@@ -46,12 +60,26 @@ def self.create_backup_tar(working_dir, backup_dir)
       return tar_file
     end
 
-    Contract String => nil
-    def restore(backup_archive)
-      @fs_restore.restore(backup_archive)
-      @db_restore.restore()
+    Contract String, String => String
+    def self.verify_hash_file(working_dir)
+      puts 'info: verifying sha256sums file'
+      Dir.chdir(working_dir)
+      `gpg --verify SHA256SUMS.sig SHA256SUMS`
+    end
+
+    Contract String, String => String
+    def self.verify_sha_file_content(working_dir)
+      puts 'info: verifying sha256sums file'
+      Dir.chdir(working_dir)
+      `sha256sum --status --strict -c SHA256SUMS`
     end
 
+#    Contract String => nil
```

Comment:
Was the main motivation for not using the `StellarCoreBackup::utils.restore` method because you needed to extract the backup prior to verification ?

Reply:
There were two motivations, the primary one you have identified, I couldn't see how to sensibly insert the gpg --verify and sha256sum --check calls in there without it obviously having been shoehorned in. The other motivation was that the backup work was done via separate commands so there was an incongruity.

Reply:
https://github.com/stellar/puppet-v4/issues/178